jinja2/lexer.py

```diff
@@ -577,6 +565,10 @@ def wrap(self, stream, name=None, filename=None):
                 token = value
             elif token == 'name':
                 value = str(value)
+                if check_ident and not value.isidentifier():
```

Comment:
This isn't sufficient guarding against python2 for some reason. It still raises an AttributeError.